angular2-jwt.ts

```diff
@@ -15,46 +15,62 @@ export interface IAuthConfig {
   noTokenScheme?:boolean;
 }
 
+// This interface should be used in the ctor of AuthHttp.
+export interface IAuthConfigOptional {
+    headerName?: string;
+    headerPrefix?: string;
+    tokenName?: string;
+    tokenGetter?: any;
+    noJwtError?: boolean;
+    globalHeaders?: Array<Object>;
+    noTokenScheme?: boolean;
+}
+
+class AuthConfigConsts {
```

Comment:
Just have them. They don't need a namespace (technically they're part of the angular2-jwt module). For the most part, they're used exactly once in the default config, so factoring them out isn't a real benefit either. It makes sense to export them so that users can reference them for their own purposes.

(While I think things like SOLID are excellent ideas and practices, they're guidelines, not dogma to be worshipped).